recipes/gn/all/conanfile.py

```diff
@@ -0,0 +1,119 @@
+        del self.info.settings.compiler
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder)
+
+    def build_requirements(self):
+        self.build_requires("ninja/1.10.2")
+
+    @contextmanager
+    def _build_context(self):
+        if self.settings.compiler == "Visual Studio":
+            with tools.vcvars(self.settings):
+                yield
+        else:
+            compiler_defaults = {}
+            if self.settings.compiler == "gcc":
+                compiler_defaults = {
+                    "CC": "gcc",
+                    "CXX": "g++",
+                    "AR": "ar",
+                    "LD": "g++",
+                }
+            elif self.settings.compiler == "clang":
+                compiler_defaults = {
+                    "CC": "clang",
+                    "CXX": "clang++",
+                    "AR": "ar",
+                    "LD": "clang++",
+                }
+            env = {}
+            for k in ("CC", "CXX", "AR", "LD"):
+                v = tools.get_env(k, compiler_defaults.get(k, None))
+                if v:
+                    env[k] = v
+            with tools.environment_append(env):
+                yield
+
+    @staticmethod
+    def _to_gn_platform(os_, compiler):
+        if tools.is_apple_os(os_):
+            return "darwin"
+        if compiler == "Visual Studio":
+            return "msvc"
+        # Assume gn knows about the os
+        return str(os_).lower()
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            with self._build_context():
+                tools.save(os.path.join("src", "gn", "last_commit_position.h"),
+                           textwrap.dedent("""\
+                                #pragma once
+                                #define LAST_COMMIT_POSITION "1"
+                                #define LAST_COMMIT_POSITION_NUM 1
+                                """))
+                conf_args = [
+                    "--no-last-commit-position",
+                    "--host={}".format(self._to_gn_platform(self.settings.os, self.settings.compiler)),
+                ]
+                if self.settings.build_type == "Debug":
+                    conf_args.append("-d")
+                self.run("python build/gen.py {}".format(" ".join(conf_args)), run_environment=True)
+                build_args = [
+                    "-C", "out",
+                    "-j{}".format(tools.cpu_count()),
+                ]
+                self.run("ninja {}".format(" ".join(build_args)), run_environment=True)
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        self.copy("gn", src=os.path.join(self._source_subfolder, "out"), dst="bin")
+        self.copy("gn.exe", src=os.path.join(self._source_subfolder, "out"), dst="bin")
+
+    def package_info(self):
+        bin_path = os.path.join(self.package_folder, "bin")
```

Comment:
Are these lines really necessary? It is my understanding that conan exposes `self.cpp_info.binpath` to PATH automatically when a package is listed as a build requirement.

Reply:
It's not needed when doing crossbuilds (`-pr:h hostprofile -pr:b buildprofile`), but I think not when using the single profile approach (when your profile contains `os_build` or `arch_build`.